modules/data.atmosphere/R/download.NOAA.R

```diff
@@ -0,0 +1,387 @@
+  # these fail to match up.
+  # for (i in 1:length(cf_var_names)) {
+  #  print(sprintf("cf / noaa   :   %s / %s", cf_var_names[[i]], noaa_var_names[[i]]))
+  #}
+  
+  noaa_data = list()
+  
+  #Setting up the loops.  num_days represents the length of time in days betwen the start and end of the data being collected.
+  num_days = lubridate::time_length(lubridate::interval(start_date, end_date), unit="day")
+  
+  #If the requested forecast data ends on the current day, we have to be careful about the time of day we ask
+  #for data.  Data is posted every 6 hours.  If this function is called at 9:00 a.m., it wouldn't make sense
+  #to request data for 12 a.m. or 6:00 p.m. - the data doesn't yet exist.
+  ends_today = FALSE
+  if (as.POSIXlt(Sys.Date()) == end_date) {
+    ends_today = TRUE
+    #The last day will be handled in its own special block of code...
+  } else {
+    num_days = num_days + 1 # ...otherwise, we need to do it in the main loop.
+  }
+  
+  #This is where the actual download of the data occurs (for most data).  This loop handles days that have already passed.  Therefore,
+  #all data is avaliable for those days, and there doesn't need to be any checking to see if the data exists or not.
+  
+  if (verbose) {
+    PEcAn.logger::logger.info("Beginning Data Download...")
+  }
+  
+  i = 1
+  max = length(cf_var_names)
+  while (i <= max && as.logical(num_days)) {
+    current_date = start_date #Incremented in the loop to fetch the proper day's data
+    
+    gefs_date = POSIX.to.GEFSDate(current_date)
+    
+    var_dat = matrix(rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0000", date=gefs_date)$data,
+                     nrow=21, ncol=1)
+    var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0600", date=gefs_date)$data)
+    var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1200", date=gefs_date)$data)
+    var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1800", date=gefs_date)$data)
+    j = 1
+    while (j < num_days) {
+      current_date = current_date + lubridate::days(1)  #increments the date by one
+      
+      gefs_date = POSIX.to.GEFSDate(current_date)
+      var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0000", date=gefs_date)$data)
+      var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "0600", date=gefs_date)$data)
+      var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1200", date=gefs_date)$data)
+      var_dat = cbind(var_dat, rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1, forecast_time = "1800", date=gefs_date)$data)
```

Comment:
I'm really not following what you're trying to do here, but I'm pretty sure it's not what we're looking for. First, because time_idx is set to 1 it seems like you're not grabbing the full forecast, just the first 6hr, which isn't what we want. We want the full timeseries, start date to the end date, which requires setting date and time_idx. Second, setting `forecast_time` requires that we need to know the `start_time`, not just the `start_date`, and find the nearest start time to the one requested. Third, the NOAA server will be much more efficient if you grab large chunks of data at once rather than grabbing a lot of little data and cbinding it together on our side.